src/migrations/20240611185509-goalStatusChanges-historical.js

```diff
@@ -0,0 +1,191 @@
+/* eslint-disable no-tabs */
+const { prepMigration } = require('../lib/migration');
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => queryInterface.sequelize.transaction(
+    async (transaction) => {
+      await prepMigration(queryInterface, transaction, __filename);
+      // Change the userName column of GoalStatusChanges to allow null.
+      await queryInterface.changeColumn('GoalStatusChanges', 'userName', {
+        type: Sequelize.STRING,
+        allowNull: true,
+      }, { transaction });
+
+      await queryInterface.sequelize.query(/* sql */`
+select create_timeseries_from_audit_log('Users');
+select create_timeseries_from_audit_log('UserRoles');
+
+WITH
+	reason_lookup AS (
```

Comment:
How were the values in reason_lookup collected? Why are we not rerunning that logic as opposed to just hardcoding it?

Reply:
it was manually generated for each of the combinations from historic data